haystack/preview/dataclasses/document.py

```diff
@@ -114,41 +122,49 @@ def _create_id(self):
         data = f"{text}{dataframe}{blob}{mime_type}{metadata}{embedding}"
         return hashlib.sha256(data.encode("utf-8")).hexdigest()
 
-    def to_dict(self):
-        """
-        Saves the Document into a dictionary.
+    def to_dict(self, flatten=True) -> Dict[str, Any]:
         """
-        return asdict(self)
+        Converts Document into a dictionary.
+        `dataframe` and `blob` fields are converted JSON serialisable types.
 
-    def to_json(self, json_encoder: Optional[Type[DocumentEncoder]] = None, **json_kwargs):
+        :param flatten: Whether to flatten `metadata` field or not. Defaults to True.
         """
-        Saves the Document into a JSON string that can be later loaded back. Drops all binary data from the blob field.
-        """
-        dictionary = self.to_dict()
-        del dictionary["blob"]
-        return json.dumps(dictionary, cls=json_encoder or DocumentEncoder, **json_kwargs)
+        data = asdict(self)
+        if (dataframe := data.get("dataframe", None)) is not None:
+            data["dataframe"] = dataframe.to_json()
+        if blob := data.get("blob", None):
```

Comment:
no need for `None` and you might want to do `if blob := data.get("blob") is not None:` for symmetry with the previous `if`